### PR TITLE
Address BigAddress UI issues

### PR DESCRIPTION
**Pull request by jubalm**
#1180 introduced some issues that this PR resolves
- [x] remove bigaddress background overlap when nested in a button
- [x] add disable copy when clicking icon
- [x] cursor and component flickering when hovering over address without copy action
- [x] change default card action to undefined instead of copy

### CODE REVIEW

**Review by KillariDev — APPROVED** [review]

File: app/ts/components/subcomponents/MultilineCard.tsx [hunk 1]
```diff
@@ -4,88 +4,128 @@ import { Tooltip, TooltipConfig } from './Tooltip.js'
...
+}
+
+const ActionableIcon = (props: ActionableIconProps) => {
 	const tooltipConfig = useSignal<TooltipConfig | undefined>(undefined)
 
 	const copyTextToClipboard = async (event: JSX.TargetedMouseEvent<HTMLButtonElement>) => {
 		event.currentTarget.blur()
 		await clipboardCopy(event.currentTarget.value)
-		tooltipConfig.value = { message: 'Copied!', x: event.clientX, y: event.clientY }
+		const copySuccessMessage = props.action === 'clipboard-copy' && props.copySuccessMessage ? props.copySuccessMessage : 'Copied!'
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/MultilineCard.tsx]: `MultilineCard` is a general use component accepts a `copySuccessMessage` and is the backbone for BigAddress where the message, `Address copied!`, is actually defined

File: app/ts/components/subcomponents/MultilineCard.tsx [hunk 2]
```diff
@@ -4,88 +4,128 @@ import { Tooltip, TooltipConfig } from './Tooltip.js'
...
 }
 
-const TextNode = ({ displayText, value }: TextNodeProps) => <data class = 'truncate text-legible' value = { value || displayText }>{ displayText }</data>
+const ActionableText = (props: ActionableTextProps) => {
+	const tooltipConfig = useSignal<TooltipConfig | undefined>(undefined)
+
+	const copyTextToClipboard = async (event: JSX.TargetedMouseEvent<HTMLButtonElement>) => {
+		event.currentTarget.blur()
+		await clipboardCopy(event.currentTarget.value)
+		tooltipConfig.value = { message: 'Copied!', x: event.clientX, y: event.clientY }
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/MultilineCard.tsx]: the value will now be dynamically set from `copySuccessMessage` in [bfa7fc8](https://github.com/DarkFlorist/TheInterceptor/pull/1190/commits/bfa7fc824b7c5c369f995bb89a922b287746e546)

File: app/ts/components/subcomponents/address.tsx [hunk 3]
```diff
@@ -58,34 +58,40 @@ type BigAddressParams = {
...
 	const noteConfig: ActionableTextProps = {
-		displayText: subTitle,
-		action: addrString && subTitle !== addrString ? renameAddressAction : undefined
+		displayText: noteText,
+		...(noteText === addressString && !params.noCopying) ? setupCopyAction : setupEditAction
+	}
+
+	const iconConfig: ActionableIconProps = {
+		icon: () => params.addressBookEntry ? <Blockie address = { params.addressBookEntry.address } /> : <></>,
+		...(!params.noCopying && addressString) ? setupCopyAction : { action: undefined }
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/address.tsx]: `action` is now flattened and exposed as props in [bfa7fc8](https://github.com/DarkFlorist/TheInterceptor/pull/1190/commits/bfa7fc824b7c5c369f995bb89a922b287746e546)

File: app/ts/components/subcomponents/MultilineCard.tsx [hunk 4]
```diff
@@ -4,88 +4,128 @@ import { Tooltip, TooltipConfig } from './Tooltip.js'
...
-		tooltipConfig.value = { message: 'Copied!', x: event.clientX, y: event.clientY }
+		const copySuccessMessage = props.action === 'clipboard-copy' && props.copySuccessMessage ? props.copySuccessMessage : 'Copied!'
+		tooltipConfig.value = { message: copySuccessMessage, x: event.clientX, y: event.clientY }
 	}
 
-	const CardIcon = icon.component
-	const defaultAction:TextAction = { label: 'Copy', icon: () => <CopyIcon />, onClick: copyTextToClipboard }
+	const CardIcon = props.icon
+	const handleClick = props.action ? props.action === 'clipboard-copy' ? copyTextToClipboard  : props.action.onClick : undefined
+	const copyValue =  props.action === 'clipboard-copy' ? props.copyValue : undefined
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/MultilineCard.tsx]: [ec3c4d5](https://github.com/DarkFlorist/TheInterceptor/pull/1190/commits/ec3c4d55659f6c51b9ae441bd6e5cfb70da6175b)

File: app/ts/components/subcomponents/MultilineCard.tsx [hunk 5]
```diff
@@ -4,88 +4,128 @@ import { Tooltip, TooltipConfig } from './Tooltip.js'
...
+	const actionHandler = props.action ? props.action === 'clipboard-copy' ? copyTextToClipboard : props.action.onClick : undefined
+	const actionButtonLabel = props.action === 'clipboard-copy' ? 'Copy' : props.action?.label || ''
 
-const ActionableText = ({ displayText, value, action }: ActionableTextProps) => {
-	const DisplayText = () => <TextNode displayText = { displayText } value = { value || displayText } />
 	return (
 		<span>
 			<DisplayText />
-			{ action !== undefined && action !== 'noaction' ? <TextAction { ...action } textNode = { DisplayText } /> : <></> }
+			<TextAction label = { actionButtonLabel } textNode = { DisplayText } icon = { actionIcon } onClick = { actionHandler } copyValue = { copyValue }  />
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/MultilineCard.tsx]: fixed in [ec3c4d5](https://github.com/DarkFlorist/TheInterceptor/pull/1190/commits/ec3c4d55659f6c51b9ae441bd6e5cfb70da6175b)

File: app/ts/components/subcomponents/MultilineCard.tsx [hunk 6]
```diff
@@ -4,88 +4,128 @@ import { Tooltip, TooltipConfig } from './Tooltip.js'
...
 		await clipboardCopy(event.currentTarget.value)
-		tooltipConfig.value = { message: 'Copied!', x: event.clientX, y: event.clientY }
+		const copySuccessMessage = props.action === 'clipboard-copy' && props.copySuccessMessage ? props.copySuccessMessage : 'Copied!'
+		tooltipConfig.value = { message: copySuccessMessage, x: event.clientX, y: event.clientY }
 	}
 
-	const CardIcon = icon.component
-	const defaultAction:TextAction = { label: 'Copy', icon: () => <CopyIcon />, onClick: copyTextToClipboard }
+	const CardIcon = props.icon
+	const handleClick = props.action ? props.action === 'clipboard-copy' ? copyTextToClipboard  : props.action.onClick : undefined
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/MultilineCard.tsx]: fixed in [ec3c4d5](https://github.com/DarkFlorist/TheInterceptor/pull/1190/commits/ec3c4d55659f6c51b9ae441bd6e5cfb70da6175b)

**Pull request merged**